sdk/core/Azure.Core.Amqp/src/AmqpMessageBody.cs

```diff
@@ -173,7 +173,41 @@ private static void ValidateAmqpPrimitive(object value, string paramName)
                 case bool:
                 case Guid:
                 case DateTime:
```

Comment:
I think that we're missing a couple of well-known types:

- `Uri`
- `DateTimeOffset`
- `TimeSpan`

Both [Service Bus](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/SerializationUtilities.cs#L40) and [Event Hubs](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/TypeExtensions.cs#L33) have logic to map these as a special case.

Reply:
I wonder if it makes sense to try and refactor to a common structure for the well-known primitives that can be used as a basis for this check as well as the conversion performed in the message converter.

Reply:
Yeah I can investigate moving this to core.amqp and using shared source to pull it into SB.

Reply:
Uri, DateTimeOffset, and Timespan are not included here though - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpMessageConverter.cs#L578-L593 or in the SB version, which is why I didn't include them for the body.

Reply:
Nevermind - I see that this is just for output from the Amqp lib.

Reply:
It isn't straightforward to refactor this as the logic is a little different within SB since it is more permissive in terms of allowing AMQP lib types within the application properties. For now, I will just add the missing types.

Reply:
Fair enough.   Thanks for taking a look.